source/distributed/shared_tria.cc

```diff
@@ -76,14 +103,85 @@ namespace parallel
                   active_halo_layer.find(cell) == active_halo_layer.end())
                 cell->set_subdomain_id(numbers::artificial_subdomain_id);
             }
+
+          // loop over all cells in multigrid hierarchy and mark artificial:
+          if (settings & construct_multigrid_hierarchy)
```

Comment:
> backwards-compatibility thing, right?

not really, it's a feature, see https://github.com/dealii/dealii/issues/1368#issuecomment-131360821

the reason for that auxiliary vector to store ids is due to the problem here https://github.com/dealii/dealii/pull/1674#issuecomment-143314553